### ISSUE TITLE
Observation room with adjustable size.

### ISSUE THREAD

**Issue by ZombiePwner1** [TeamSpen210/BEE2-items#1165]
This was a cut feature but I'm sure it can be reimplemented.

**Comment by Byzarru** [comment]
I still want to make a new suggestion, but I think there should be "prop" items that place things like chairs, desks, and computers in PTI maps. With this, you can at least fake the observation room.

What could probably be done in this case is a glass-sizable item that recesses in the wall, using a bit of math to position desks, lights, and doors accordingly.

**Comment by ZombiePwner1** [comment]
It can be a regular observation room. But I don't know the exact model since this was removed from the puzzlemaker and there's no icon for it.

**Comment by Luke18033** [comment]
this would probably be hard - if not impossible - to implement. There would probably just be three different sizes with Button Type or something.

**Comment by TeamSpen210** [comment]
I'm planning on implementing this. It'll be an item which embeds a 3x1 tall area, so you can place multiple side-by-side. Any sides of this open to the map will become windows, and inside walls, doors and random furniture will be generated.

**Comment by Flametusk** [comment]
Could it be possible to have a vertical variant? Possibly allowing for stairs and things in the observation rooms?

**Comment by TeamSpen210** [comment]
How would that work? I'm not too sure how a vertical obs room would work.

**Comment by Luke18033** [comment]
@ZombiePwner1 I don't think Valve ever intended to include this.

**Comment by ZombiePwner1** [comment]
No if u read the puzzle creator wiki on the portal wiki you will see down the list with the unused items and one of them was an observation room with adjustable size. Take a look at the wiki: http://theportalwiki.com/wiki/Puzzle_Creator

**Comment by Luke18033** [comment]
Oh I never noiticed that.

**Comment by ZombiePwner1** [comment]
Read that wiki it explains it.

**Comment by Luke18033** [comment]
No it doesn't... it just says "found in game files". Where in the game files? Was there editoritems code? A model?

**Comment by ZombiePwner1** [comment]
Most likely editoritems code.

**Comment by TeamSpen210** [comment]
It's potentially from this [screenshot](http://theportalwiki.com/wiki/File:Portal_2_editor_announcement_02.jpg), which shows a 4x1 observation room.

**Comment by Luke18033** [comment]
How do we know it's resizeable though? It could have just been a 4x1 Observation Room item.

**Comment by Byzarru** [comment]
> Really, the entire build-a-room should have this kind of setup with room sizes and brightness.

 (#1322)

**Comment by ZombiePwner1** [comment]
Actually @Luke18033 i think this image explains it since i found this in the games files.

![observationroom](https://cloud.githubusercontent.com/assets/18079316/22392642/910e8342-e536-11e6-9f55-8ae3b3d7f0c1.png)

**Comment by Luke18033** [comment]
Where?

**Comment by ZombiePwner1** [comment]
What do you mean where?

**Comment by Luke18033** [comment]
Where did you find it in the files?

**Comment by Endermage77** [comment]
And what kind of file is it?

**Comment by TeamSpen210** [comment]
It's one of the VTFs for the palette icons, but not used.

**Comment by ZombiePwner1** [comment]
@Luke18033 It's located in portal2_dlc2/materials/puzzlemaker

**Comment by ZombiePwner1** [comment]
@TeamSpen210 The unused icon could be the image used for the adjustable observation room like i showed above.

**Comment by TeamSpen210** [comment]
I don't think so. It's kinda got a different aesthetic, so it wouldn't really fit very well. I'd probably just use the normal obs room icon, with a 4-way arrow to show it can resize like Cutout Tiles.

**Comment by Konclan** [comment]
"What I think I'd like to try and do is make a customization observation room, which would be a 1x1 section you place next to each other, then generating a random room design." -TeamSpen210
